modules/mailer/mailer.go

```diff
@@ -195,6 +195,47 @@ func (s *Sender) Send(from string, to []string, msg io.WriterTo) error {
 	return client.Quit()
 }
 
+func (s *Sender) sendSendmail(from string, to []string, msg io.WriterTo) error {
```

Comment:
This should really be implemented as a separate `Sender` instead, this way you don't have to do checks for every single sent email...

Config-reading and setup could be done here: https://github.com/go-gitea/gitea/pull/355/files#diff-bdf9b0a2ed1413ee986db72787e5b5d7R265